installers/osie/main.go

```diff
@@ -4,87 +4,122 @@ import (
 	"context"
 	"strings"
 
+	"github.com/packethost/pkg/env"
+	"github.com/pkg/errors"
 	"github.com/tinkerbell/boots/conf"
+	"github.com/tinkerbell/boots/installers"
 	"github.com/tinkerbell/boots/ipxe"
 	"github.com/tinkerbell/boots/job"
 	"go.opentelemetry.io/otel/trace"
 )
 
-type Installer struct{}
+var (
+	osieURL     = conf.MirrorBaseURL + "/misc/osie"
+	workflowURL = conf.MirrorBaseURL + "/workflow"
 
-// DefaultHandler determines the ipxe boot script to be returned.
-// If the job has an instance with Rescue true, the rescue boot script is returned.
-// Otherwise the installation boot script is returned.
-func (i Installer) DefaultHandler() job.BootScript {
-	return func(ctx context.Context, j job.Job, s ipxe.Script) ipxe.Script {
-		if j.Rescue() {
-			return i.rescue()(ctx, j, s)
+	staticTinkArgs []string
+)
+
+type installer struct{}
+
+func Installer() job.BootScripter {
+	if env.Get("DATA_MODEL_VERSION") != "1" {
+		return installer{}
+	}
+
+	require := func(key string) string {
+		val := env.Get(key)
+		if val == "" {
+			installers.Logger("osie").With("key", key).Fatal(errors.New("invalid key"))
 		}
 
-		return i.install()(ctx, j, s)
+		return val
 	}
-}
 
-// rescue generates the ipxe boot script for booting into osie in rescue mode
-func (i Installer) rescue() job.BootScript {
-	return func(ctx context.Context, j job.Job, s ipxe.Script) ipxe.Script {
-		s.Set("action", "rescue")
-		s.Set("state", j.HardwareState())
+	staticTinkArgs = []string{
+		"grpc_authority=" + require("TINKERBELL_CERT_URL"),
+		"grpc_cert_url=" + require("TINKERBELL_GRPC_AUTHORITY"),
```

Comment:
Mix up here:
```suggestion
		"grpc_authority=" + require("TINKERBELL_GRPC_AUTHORITY"),
		"grpc_cert_url=" + require("TINKERBELL_CERT_URL"),
```

Also:
* Isn't `TINKERBELL_CERT_URL` deprecated/gone and replaced with `TINKERBELL_TLS=bool`?
* You'll need a way to plumb down the tink API for the Kubernetes backend data model

Reply:
oops I've *still* forgotten about k8s backend, have to update that too.

Reply:
I kept CERT_URL and left TLS out originally because it didn't seem like "spring cleaning", but after you mentioned it I think its fine, it should have been done when those respective PRs landed in tink so I can see this being "dirty" in need of cleaning ... if you squint :D. So I've fixed this now.

Reply:
@micahhausler the kubernetes backend pr (#250) isn't merged yet.